program-runtime/src/loaded_programs.rs

```diff
@@ -2386,6 +2375,44 @@ mod tests {
         assert!(match_missing(&missing, &program3, true));
     }
 
+    #[test]
+    fn test_unloaded() {
+        let mut cache = new_mock_cache::<TestForkGraph>();
+        for loaded_program_type in [
+            LoadedProgramType::FailedVerification(cache.environments.program_runtime_v1.clone()),
+            LoadedProgramType::Closed,
+            LoadedProgramType::DelayVisibility,
```

Comment:
This can't go in the cache right? I would comment it and explicitly say that. It's weird if tests test things which aren't supposed to happen because then you can't use tests to learn an API.